validate.C

```diff
@@ -291,63 +289,98 @@ void calomet(TString algo, TString var, bool doLog10 = false){
   plotvar(v);
 }
 
-void met(TString algo, TString var, bool doLog10 = false){
-  TString v;
-  if (doLog10) v ="log10(recoMETs_"+algo+"__"+recoS+".obj."+var+"())";
-  else v = "recoMETs_"+algo+"__"+recoS+".obj."+var+"()";
+void met(TString var, TString cName = "tcMet_", TString tName = "recoMETs_",  bool notafunction=false){
+  TString v=notafunction ? tName+cName+"_"+recoS+".obj."+var:
+    tName+cName+"_"+recoS+".obj."+var+"()";
   plotvar(v);
 }
 
-void tau(TString algo, TString var){
-  TString v="recoPFTaus_"+algo+"__"+recoS+".obj."+var+"()";
+void metVars(TString cName = "tcMet_", TString tName = "recoMETs_") {
+  met("pt",cName,tName);
+  met("px",cName,tName);
+  met("py",cName,tName);
+  met("eta",cName,tName);
+  met("phi",cName,tName);
+  met("significance",cName,tName);
+}
+
+void tau(TString var, TString cName = "hpsPFTauProducer_", TString tName = "recoPFTaus_",  bool notafunction=false){
+  TString v=notafunction ? tName+cName+"_"+recoS+".obj."+var:
+    tName+cName+"_"+recoS+".obj."+var+"()";
   plotvar(v);
 }
 
-void photon(TString var, TString cName = "photons_", bool notafunction=false){
-  TString v= notafunction ? "recoPhotons_"+cName+"_"+recoS+".obj."+var :
-    "recoPhotons_"+cName+"_"+recoS+".obj."+var+"()" ;
+void tauVars(TString cName = "hpsPFTauProducer_", TString tName = "recoPFTaus_"){
+  plotvar(tName+cName+"_"+recoS+".obj@.size()");
+  tau("energy",cName,tName);
+  tau("et",cName,tName);
+  tau("eta",cName,tName);
+  tau("phi",cName,tName);
+  if (tName!="patTaus_") tau("emFraction",cName,tName);
```

Comment:
is it going to crash if emFraction is tried on patTaus?
It's easier to maintain without the extra ifs